src/auth/esritoken/CMakeLists.txt

```diff
@@ -1,38 +1,59 @@
 set(AUTH_ESRI_TOKEN_SRCS
   qgsauthesritokenmethod.cpp
-  qgsauthesritokenedit.cpp
 )
+if (WITH_GUI)
```

Comment:
I am not a huge fan of this.
With further developments, this is most likely that someone will introduce an include of the gui in the core part.
It's a minimal effort to split this into 2 distinct targets `auth` and `auth_gui` right now and will avoid headaches (after merging PRs breaking the build of core only).

Reply:
@3nids , I've templated this from the way we handle data providers here. Is there a good reason here to diverge from that? (I'm open minded, merely asking to get educated :) )